torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -271,12 +271,12 @@ std::vector<at::Tensor> ShardingUtil::ShardTensor(
                               (tensor.sizes()[2] % tile_shape[2] != 0);
         shard = tensor.index(
             {at::indexing::Slice(
-                 (core / tile_shape[1] / tile_shape[2]) * x_partition,
-                 (core / tile_shape[1] / tile_shape[2] + 1) * x_partition),
-             at::indexing::Slice((core / tile_shape[2]) * y_partition,
-                                 (core / tile_shape[2] + 1) * y_partition),
-             at::indexing::Slice((core % tile_shape[2]) * z_partition,
-                                 (core % tile_shape[2] + 1) * z_partition)});
+                 (i / tile_shape[1] / tile_shape[2]) * x_partition,
```

Comment:
why change from `core` to `i` here?

Reply:
To enable custom device assignment, so the i-th shard is indexed by core, which can be a custom device order.